nanoFramework.System.Net/Sockets/Socket.cs

```diff
@@ -749,7 +789,15 @@ public int Receive(byte[] buffer, int offset, int size, SocketFlags socketFlags)
                 throw new ObjectDisposedException();
             }
 
-            return NativeSocket.recv(this, buffer, offset, size, (int)socketFlags, m_recvTimeout);
```

Comment:
Again here. This is supposed to throw exceptions on errors. See [here](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.receive?view=netframework-4.8#System_Net_Sockets_Socket_Receive_System_Byte___System_Int32_System_Int32_System_Net_Sockets_SocketFlags_).

It's up to the caller to deal with these.

Reply:
SocketException now being thrown